mmrazor/models/losses/rkd.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..builder import LOSSES
+
+
+@LOSSES.register_module()
+class RelationalKD(nn.Module):
+    """PyTorch version of `Relational Knowledge Distillation.
+
+    <https://arxiv.org/abs/1904.05068>`_.
+    Args:
+        loss_weight_d (float): Weight of distance-wise distillation loss.
+            Defaults to 25.0.
+        loss_weight_a (float): Weight of angle-wise distillation loss.
+            Defaults to 50.0.
+        l2_norm (bool): Whether to normalize the model predictions before
+            calculating the loss. Defaults to True.
+    """
+
+    def __init__(self, loss_weight_d=25.0, loss_weight_a=50.0, l2_norm=True):
+        super(RelationalKD, self).__init__()
+        self.loss_weight_d = loss_weight_d
+        self.loss_weight_a = loss_weight_a
+        self.l2_norm = l2_norm
+
+    def euclidean_distance(self, pred, squared=False, eps=1e-12):
+        """Calculate the Euclidean distance between the two examples in the
+        output representation space.
+
+        Args:
+            pred (torch.Tensor): The prediction of the teacher or student with
+                shape (N, C).
+            squared (bool): Whether to calculate the squared Euclidean
+                distance. Defaults to False.
+            eps (float): The minimum Euclidean distance between the two
+                examples. Defaults to 1e-12.
+        """
+        pred_square = pred.pow(2).sum(dim=-1)
```

Comment:
It is better to add some doc-string to describe the process of shape change

Reply:
Fixed, thx!